Bottom: 57443424e966f87819d9fcaf8b4d3735be50661d
Top:    4c64dd84e7e54d54741cba5aad805e713a01c31f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-03 08:45:03 -0500

Always update configuration on startup

The entrypoint script will always copy updated certificates and incorporate
updated configuration on container startup.


---
diff --git a/docker/docker-exim4/docker-entrypoint.sh b/docker/docker-exim4/docker-entrypoint.sh
index 8d9d35cb..3c3e0a26 100755
--- a/docker/docker-exim4/docker-entrypoint.sh
+++ b/docker/docker-exim4/docker-entrypoint.sh
@@ -20,27 +20,25 @@ chown -R root:mail /var/mail
 
 case ${1} in
     exim4)
-        if   [[ -f ${CONFIG_PATH}/update-exim4.conf.conf ]] \
-          && [[ -f ${CONFIG_PATH}/exim4.conf.template ]] \
-          && [[ ! -f ${CONFIG_PATH}/exim4.conf ]]
-        then
-            # set the hostname
-            echo "${HOSTNAME}" > /etc/mailname
-            # copy certificate
-            [ -e /etc/ssl/private/exim.crt ] \
-                && rm -f ${CONFIG_PATH}/exim.crt \
-                && cp $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt \
-                && chmod 640 ${CONFIG_PATH}/exim.crt \
-                && chown root:Debian-exim ${CONFIG_PATH}/exim.crt
-            # copy key
-            [ -e /etc/ssl/private/exim.key ] \
-                && rm -f ${CONFIG_PATH}/exim.key \
-                && cp $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key \
-                && chmod 640 ${CONFIG_PATH}/exim.key \
-                && chown root:Debian-exim ${CONFIG_PATH}/exim.key
-            # run the debian utility to update/generate the configuration
-            $(which update-exim4.conf) --verbose
-        fi
+        # set the hostname
+        echo "${HOSTNAME}" > /etc/mailname
+
+        # copy certificate
+        [ -e /etc/ssl/private/exim.crt ] \
+            && rm -f ${CONFIG_PATH}/exim.crt \
+            && cp $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt \
+            && chmod 640 ${CONFIG_PATH}/exim.crt \
+            && chown root:Debian-exim ${CONFIG_PATH}/exim.crt
+
+        # copy key
+        [ -e /etc/ssl/private/exim.key ] \
+            && rm -f ${CONFIG_PATH}/exim.key \
+            && cp $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key \
+            && chmod 640 ${CONFIG_PATH}/exim.key \
+            && chown root:Debian-exim ${CONFIG_PATH}/exim.key
+
+        # run the debian utility to update/generate the configuration
+        $(which update-exim4.conf) --verbose
         
         # Check configuration
         if ! /usr/sbin/exim4 -bV > /dev/null ; then
